### ISSUE TITLE
Battery Status & Fn shortcuts issue after waking from sleep

### ISSUE THREAD

**Issue by SukkaW** [samtheprogram/efi-e480-20kn-8th#1]
I am using ThinkPad E480 with my own EFI. I have noticed that the battery stat won't update after waking from sleep:

- If the laptop is has no AC plugged in before sleep, then it will shows uncharged even when AC is plugged in after waking.
- If the laptop is has AC plugged in before sleep, then it will shows charging even when AC is off after waking.
- And, Fn shortcuts isn't functioning as well (with FnLock light on) after waking from sleep.

It seems that those two problem are connected: either Fn & Battery status work at the same time, or the two neither work at the same time.

See also https://github.com/dandepeched/Hackintosh-ThinkPad-E580/issues/3

I am wondering if you have the same issue (since you already said `Lid close not working after sleep`).

**Comment by sammoore** [comment]
@SukkaW Yup, I'm also experiencing the battery status mismatch after sleep, and the Fn keys not working properly with FnLock on. From all DSDT debugging I did, I suspect at least a couple of these are related to each other / have the same cause... I always looked at the battery status mismatch separately from the FnLock issue, because IIRC the FnLock issue was consistent with every sleep, but the battery issue _seemed_ to only occur if I changed charging states (e.g. charging -> unplugged) during sleep.

The issue over at the E580 repo has some interesting info. I haven't looked at the WAK methods in almost a year, but would be interested in revisiting it. My machine is still running Mojave so there is merit to it as well 😄.

**Comment by SukkaW** [comment]
Battery status & Fn Function are both related with EC (More specifically, EC event become untriggerable after S3 sleep).

Currently I am using https://github.com/zhen-zen/YogaSMC to read some EC value, to see if it changes after wake.

**Comment by Juandcode** [comment]
did you fix that problem?

**Comment by andreszerocross** [comment]
> Battery status & Fn Function are both related with EC (More specifically, EC event become untriggerable after S3 sleep).
> 
> Currently I am using https://github.com/zhen-zen/YogaSMC to read some EC value, to see if it changes after wake.

There is same linux problem with E14,, https://github.com/masksshow/Thinkpad-E14-15-AMD-Gen-2-FIX

I have check the script but it only change H8DR from 0x00 to One.

 I have tried it in Lenovo E14 but problem is still there

**Comment by farseenmanekhan1232** [comment]
is this resolved ?

**Comment by nalakawula** [comment]
Came here to ask, is it fixable?

**Comment by sammoore** [comment]
I no longer use this laptop, and it was "good enough" for me while I had it, so I never resolved it.

If I were you, I would look into [SukkaW's comment](https://github.com/sammoore/efi-e480-20kn-8th/issues/1#issuecomment-689253487) and hack something together. Just not something I can spend any time on.

Happy to merge anything into this repo if someone has a fix, no matter how hacky it might be.
